packages/nameguard-sdk/src/index.ts

```diff
@@ -376,15 +376,15 @@ export interface ConsolidatedNameGuardReport extends ConsolidatedReport {
 /**
  * NameGuard report that contains the full results of all `checks` on all `labels` in a name.
  */
-export interface NameGuardReport extends ConsolidatedNameGuardReport {
+export interface AbstractNameGuardReport extends ConsolidatedNameGuardReport {
   /* The results of all checks performed by NameGuard on `name`. */
   checks: CheckResult[];
 
   /** Details of the inspection of all labels in `name`.
     *
     * `null` if `name` is uninspected
   */
-  labels: LabelGuardReport[] | null;
+  labels?: LabelGuardReport[];
 
   /**
    * The name considered to be the canonical form of the analyzed `name`.
```

Comment:
Here's a suggestion for the line below this line. GitHub won't let me comment on the line I want to comment.

```suggestion
   * `undefined` if and only if the canonical form of `name` is considered to be undefined or `name` was uninspected and this is an ``UninspectedNameGuardReport`.
```